packages/bbui/src/Typography/Heading.svelte

```diff
@@ -2,10 +2,10 @@
   import "@spectrum-css/typography/dist/index-vars.css"
 
   // Sizes
-  export let size = "M"
-  export let textAlign = undefined
-  export let noPadding = false
-  export let weight = "default" // light, heavy, default
+  export let size: "XS" | "S" | "M" | "L" = "M"
+  export let textAlign: string | undefined = undefined
```

Comment:
I think that the `| undefined` is not required, as it gets inferred by the `= undefined`

Reply:
You got me hopeful as I have done this in loads of places and would love to remove it!

But it does seem required. If I remove it:
![image](https://github.com/user-attachments/assets/69a8d176-fed4-4489-977a-d1a3aac6918a)

Reply:
Mm.. I just used in here: https://github.com/Budibase/budibase/pull/15384/files#diff-615c4b1d1bda62d5c443c2b06e9226cb16b173146419c01e458d72d1dafe7383

Reply:
Oh, I see. It's caused to adding the default tsconfig. The `strictNullChecks` is the culprit

Reply:
Ah I think I understand the difference. In this case `textAlign` actually is optional, so I want to type is as possibly undefined. If i change `strictNullChecks` to false then it lets me set it as undefined, but it is typed as `string` only which is incorrect for this use case and potentially dangerous (unless all our code was TS, then it would be easy to spot any errors). 

I'll keep this as-is, because I think it's more correct for this use case, in that it actually is optional 👍